### PR TITLE
Switch to go 1.22

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind ci

#### What this PR does / why we need it:
Fix to the latest go version and adjust the failing test.
#### Which issue(s) this PR fixes:

None
#### Special notes for your reviewer:
None
#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Comment by saschagrunert** [comment]
@rphillips @haircommander PTAL

File: pkg/client/client_test.go [hunk 1]
```diff
@@ -707,7 +711,8 @@ var _ = Describe("JSONLogger", func() {
 						Path: tr.logPath(),
 					}},
 				})
-				Expect(err).To(HaveOccurred())
+				Expect(err).NotTo(HaveOccurred())
+				tr.startContainer(sut)
```
Comment (haircommander) [inline pkg/client/client_test.go:715]: it's not clear to me why this switched

Comment (saschagrunert) [reply pkg/client/client_test.go:715]: I changed the container above to actually be valid. Fun fact is that the for loop `for _, terminal := range []bool{true, false} {` in go 1.21 (ginkgo related?) does not work for `true`, but it works in go 1.22.

**Pull request merged**